src/controller/CHIPDeviceController.cpp

```diff
@@ -2311,6 +2312,10 @@ CHIP_ERROR DeviceCommissioner::ParseICDInfo(ReadCommissioningInfo & info)
     }
 
     ReturnErrorOnFailure(err);
+    if (!(info.icd.isICD))
+    {
+        return CHIP_NO_ERROR;
+    }
```

Comment:
If the device is not and ICD, won't we error out just before since the cluster won't be present?
If the cluster, is present we will always be an ICD no?

Reply:
update icdICD with local boolean. In previous design talk, we decide to use featuremap for ICDManagement for ICD cluster check to reduce several unnecessary check. 

So if this featureMap of the ICDManagementCluster exists, we think it is ICD device, if not exists, we just return CHIP_NO_ERROR here. thanks